server/expression/any.go

```diff
@@ -0,0 +1,359 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if rightInterface == nil {
+		return nil, nil
+	}
+
+	rightValues, ok := rightInterface.([]any)
+	if !ok {
+		return nil, fmt.Errorf("%T: expected right child to return `%T` but returned `%T`", a, []any{}, rightInterface)
+	}
+	if len(rightValues) == 0 {
+		return nil, nil
+	}
+
+	// Next we'll assign our evaluated values to the expressions that the comparison function reference
+	a.staticLiteral.value = left
+	for _, rightValue := range rightValues {
+		a.arrayLiteral.value = rightValue
+		result, err := a.compFunc.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+		if result.(bool) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// Eval implements the Expression interface.
+func (a *AnyExpr) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	left, err := a.leftExpr.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if a.subqueryAnyExpr != nil {
+		return a.subqueryAnyExpr.eval(ctx, a.subOperator, row, left)
+	}
+
+	if a.expressionAnyExpr != nil {
+		return a.expressionAnyExpr.eval(ctx, row, left)
+	}
+
+	return nil, fmt.Errorf("%T: cannot Eval as it has not been fully resolved", a)
+}
+
+// WithChildren implements the Expression interface.
+func (a *AnyExpr) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 2 {
+		return nil, sql.ErrInvalidChildrenNumber.New(a, len(children), 2)
+	}
+
+	anyExpr := &AnyExpr{
+		leftExpr:    children[0],
+		rightExpr:   children[1],
+		subOperator: a.subOperator,
+		name:        a.name,
+	}
+
+	if sub, ok := children[1].(*plan.Subquery); ok {
+		return subqueryWithChildren(anyExpr, sub)
+	}
+
+	return expressionWithChildren(anyExpr)
+}
+
+// WithResolvedChildren implements the Expression interface.
+func (a *AnyExpr) WithResolvedChildren(children []any) (any, error) {
+	if len(children) != 2 {
+		return nil, fmt.Errorf("invalid vitess child count, expected `2` but got `%d`", len(children))
+	}
+	left, ok := children[0].(sql.Expression)
+	if !ok {
+		return nil, fmt.Errorf("expected vitess child to be an expression but has type `%T`", children[0])
+	}
+	right, ok := children[1].(sql.Expression)
+	if !ok {
+		return nil, fmt.Errorf("expected vitess child to be an expression but has type `%T`", children[1])
+	}
+	return a.WithChildren(left, right)
+}
+
+// String implements the fmt.Stringer interface.
+func (a *AnyExpr) String() string {
+	if a.leftExpr == nil || a.rightExpr == nil {
+		return fmt.Sprintf("? %s (?)", a.name)
+	}
+	return fmt.Sprintf("%s %s (%s)", a.leftExpr, a.name, a.rightExpr)
+}
+
+// DebugString implements the Expression interface.
+func (a *AnyExpr) DebugString() string {
+	return fmt.Sprintf("%s %s (%s)", sql.DebugString(a.leftExpr), a.name, sql.DebugString(a.rightExpr))
+}
+
+// subqueryWithChildren resolves the comparison functions for a plan.Subquery.
+func subqueryWithChildren(anyExpr *AnyExpr, sub *plan.Subquery) (sql.Expression, error) {
```

Comment:
Maybe name this one something like `anySubqueryWithChildren`, since the name is fairly general when it only works for `AnyExpr`.